server.go

```diff
@@ -186,6 +235,248 @@ func main() {
+	if err != nil {
+		return
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", webflowToken))
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("accept-version", "1.0.0")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return
+	} else if res.StatusCode < 200 || res.StatusCode >= 300 {
+		c.Logger().Debug(req)
+		var responseBody []byte
+		_, _ = res.Body.Read(responseBody)
+		c.Logger().Debug(responseBody)
+		// return a real error to the caller to indicate failure
+		errMsg := fmt.Sprintf(msgUpstreamParticipantUpdateError, res.Status)
+		if errContext := c.String(http.StatusInternalServerError, errMsg); errContext != nil {
+			err = errContext
+		} else {
+			err = fmt.Errorf(errMsg)
+		}
+	}
+
+	return
+}
+
+var PARTICIPATING_ORGS = map[string]bool{
+	"thanos-io":          true,
+	"serverlessworkflow": true,
+	"chaos-mesh":         true,
+	"cri-o":              true,
+	"openebs":            true,
+	"buildpacks":         true,
+	"schemahero":         true,
+}
+
+func validScore(owner string, name string, user string) bool {
+	// check if repo is in participating set
+	if !PARTICIPATING_ORGS[owner] {
+		return false
+	}
+
+	// Check if participant is registered
+	sqlQuery := `SELECT
+		participants.Id
+		FROM participants
+		WHERE participants.GitHubName = $1`
+	row := db.QueryRow(sqlQuery, user)
+	var id string
+	err := row.Scan(&id)
+	if err != nil {
+		return false
+	}
+
+	return true
+}
+
+func scorePoints(msg scoring_message) (points int) {
+	points = 0
+	scored := 0
+
+	for bugType, count := range msg.BugCounts {
+		sqlQuery := `SELECT pointValue FROM bugs WHERE category = $1`
+		row := db.QueryRow(sqlQuery, bugType)
+		var value int = 1
+		// @TODO handle possible row scan error below
+		row.Scan(&value)
+		points += count * value
+		scored += count
+	}
+
+	// add 1 point for all non-classified fixed bugs
+	if scored < msg.TotalFixed {
+		points += msg.TotalFixed - scored
+	}
+
+	return
+}
+
+const sqlUpdateParticipantScore = `UPDATE participants 
+		SET Score = Score + $1 
+		WHERE GitHubName = $2 
+		RETURNING UpstreamId, Score`
+
+func updateParticipantScore(c echo.Context, username string, delta int) (err error) {
+	var id string
+	var score int
+	row := db.QueryRow(sqlUpdateParticipantScore, delta, username)
+	err = row.Scan(&id, &score)
+	if err != nil {
+		return
+	}
+
+	err = upstreamUpdateScore(c, id, score)
+	return
+}
+
+func newScore(c echo.Context) (err error) {
+	var alert scoring_alert
+	err = json.NewDecoder(c.Request().Body).Decode(&alert)
+	if err != nil {
+		return
+	}
+
+	c.Logger().Debug(alert)
+
+	for _, rawMsg := range alert.RecentHits {
+		var msg scoring_message
+		err = json.Unmarshal([]byte(rawMsg), &msg)
+		if err != nil {
+			return
+		}
+		c.Logger().Debug(msg)
+
+		// if this particular entry is not valid, ignore it and continue processing
+		if !validScore(msg.RepoOwner, msg.RepoName, msg.TriggerUser) {
```

Comment:
`msg.RepoName` parameter not used

```suggestion
		if !validScore(msg.RepoOwner, msg.TriggerUser) {
```